tests/ci/ci.py

```diff
@@ -0,0 +1,665 @@
+                state=job_status.status,
+                target_url=job_status.report_url,
+                description=f"Reused from [{job_status.pr_num}-{job_status.sha[0:8]}]: {job_status.description}",
+                context=get_check_name(job, batch=batch, num_batches=num_batches),
+            )
+            print(f"GH status re-created from file [{success_flag_name}]")
+
+    with concurrent.futures.ThreadPoolExecutor() as executor:
+        futures = []
+        for job in job_digests:
+            if is_build_job(job):
+                # no GH status for build jobs
+                continue
+            digest = job_digests[job]
+            num_batches = CI_CONFIG.get_job_config(job).num_batches
+            for batch in range(num_batches):
+                future = executor.submit(
+                    run_create_status, job, digest, batch, num_batches
+                )
+                futures.append(future)
+        done, _ = concurrent.futures.wait(futures)
+        for future in done:
+            try:
+                _ = future.result()
+            except Exception as e:
+                raise e
+    print("Going to update overall CI report")
+    set_status_comment(commit, pr_info)
+    print("... CI report update - done")
+
+    # clean up
+    ci_files = list(temp_path.glob("*.ci"))
+    for file in ci_files:
+        file.unlink()
+
+
+def main() -> int:
+    exit_code = 0
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    args = parse_args(parser)
+
+    if args.mark_success or args.pre or args.post or args.run:
+        assert args.infile, "Run config must be provided via --infile"
+        assert args.job_name, "Job name must be provided via --job-name"
+
+    indata: Optional[Dict[str, Any]] = None
+    if args.infile:
+        indata = (
+            json.loads(args.infile)
+            if not os.path.isfile(args.infile)
+            else json.load(open(args.infile))
+        )
+        assert indata and isinstance(indata, dict), "Invalid --infile json"
+
+    result: Dict[str, Any] = {}
+    s3 = S3Helper()
+
+    if args.configure:
+        GR = GitRunner()
+        pr_info = PRInfo()
+
+        docker_data = {}
+        git_ref = GR.run(f"{GIT_PREFIX} rev-parse HEAD")
+
+        # if '#no-merge-commit' is set in commit message - set git ref to PR branch head to avoid merge-commit
+        if pr_info.number != 0:
+            message = GR.run(f"{GIT_PREFIX} log {pr_info.sha} --format=%B -n 1")
+            if "#no-merge-commit" in message and CI:
+                GR.run(f"{GIT_PREFIX} checkout {pr_info.sha}")
+                git_ref = GR.run(f"{GIT_PREFIX} rev-parse HEAD")
+                print(
+                    "#no-merge-commit is set in commit message - Setting git ref to PR branch HEAD to not use merge commit"
+                )
+
+        # let's get CH version
+        version = get_version_from_repo(git=Git(True)).string
```

Comment:
to make it correct, we need as well generated contributors and cmake versions file

and it will save some time, for sure. However, after that, we still need to preserve some jobs with a full checkout, to check the logic